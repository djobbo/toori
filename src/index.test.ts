import { createWorkflow } from '.'

describe('create workflow', () => {
  it('should create a workflow', () => {
    const workflow = createWorkflow('Build and Deploy Docs')
      .on(['push', { branches: ['master'] }])
      .job('build-and-deploy', {
        concurrency: 'ci-${{ github.ref }}',
        runsOn: 'ubuntu-latest'
      }, (job) => {
        job
          .use('Checkout üõéÔ∏è', 'actions/checkout@v3')
          .use('Setup pnpm', 'pnpm/action-setup@v2')
          .use('Setup Node', 'actions/setup-node@v3', { with: { nodeVersion: '16' } })
          .step('Install and Build üîß', (step) => {
            step
              .run('pnpm ci')
              .run('pnpm docs:build')
          })
          .use('Deploy üöÄ', 'JamesIves/github-pages-deploy-action@v4.2.5', {
            with: {
              branch: 'gh-pages',
              folder: 'docs/guide/.vuepress/dist'
            }
          })
      })

    expect(workflow.toYAML()).toMatchSnapshot()
  })

  it('should create a workflow 2', () => {
    const workflow = createWorkflow('Publish dev')
      .on('workflow_dispatch', ['push', { branches: ['master'] }])
      .job('npm', {
        runsOn: 'ubuntu-latest',
        if: "github.repository_owner == 'djobbo'"
      }, (job) => {
        job
          .use('Checkout üõéÔ∏è', 'actions/checkout@v3')
          .use('Setup pnpm', 'pnpm/action-setup@v2')
          .use('Setup Node', 'actions/setup-node@v3', {
            with: {
              nodeVersion: '16',
              registryUrl: 'https://registry.npmjs.org/',
              cache: 'pnpm',
            }
          })
          .use('Turbo cache', 'actions/cache@v2', {
            id: 'turbo-cache',
            with: {
              path: '.turbo',
              key: 'turbo-${{ github.job }}-${{ github.ref_name }}-${{ github.sha }}',
              restoreKeys: 'turbo-${{ github.job }}-${{ github.ref_name }}-'
            }
          })
          .step('Publish', {
            run: 'pnpx zx ./scripts/publish-dev.mjs', env: {
              NODE_AUTH_TOKEN: '${{ secrets.NPM_PUBLISH_TOKEN }}'
            }
          })
      })
    console.log(workflow.toYAML())

    expect(workflow.toYAML()).toMatchSnapshot()

  })
})