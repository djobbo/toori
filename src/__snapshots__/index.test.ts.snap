// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`create workflow should create a workflow 1`] = `
"name: Build and Deploy Docs
on:
  - push:
      branches:
        - master
jobs:
  build-and-deploy:
    concurrency: ci-\${{ github.ref }}
    runs-on: ubuntu-latest
    steps:
      - name: Checkout 🛎️
        uses: actions/checkout@v3
      - name: Setup pnpm
        uses: pnpm/action-setup@v2
      - name: Setup Node
        uses: actions/setup-node@v3
        with:
          node-version: '16'
      - name: Install and Build 🔧
        run: |-
          pnpm ci
          pnpm docs:build
      - name: Deploy 🚀
        uses: JamesIves/github-pages-deploy-action@v4
        with:
          branch: gh-pages
          folder: docs/guide/.vuepress/dist
"
`;

exports[`create workflow should create a workflow 2 1`] = `
"name: Publish dev
on:
  - workflow_dispatch
  - push:
      branches:
        - master
jobs:
  npm:
    runs-on: ubuntu-latest
    if: github.repository_owner == 'djobbo'
    steps:
      - name: Checkout 🛎️
        uses: actions/checkout@v3
      - name: Setup pnpm
        uses: pnpm/action-setup@v2
      - name: Setup Node
        uses: actions/setup-node@v3
        with:
          node-version: '16'
          registry-url: https://registry.npmjs.org/
          cache: pnpm
      - name: Turbo cache
        uses: actions/cache@v3
        id: turbo-cache
        with:
          path: .turbo
          key: turbo-\${{ github.job }}-\${{ github.ref_name }}-\${{ github.sha }}
          restore-keys: turbo-\${{ github.job }}-\${{ github.ref_name }}-
      - name: Publish
        env:
          NODE_AUTH_TOKEN: \${{ secrets.NPM_PUBLISH_TOKEN }}
"
`;
